### DISCUSSION TITLE
Cannot find module 'next/dist/lib/metadata/types/metadata-interface.js' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by Shahekta51995** [vercel/next.js#53288] [Help] [answered]
### Summary

I am getting this error.
.next/types/app/layout.ts:3:40
Type error: Cannot find module 'next/dist/lib/metadata/types/metadata-interface.js' or its corresponding type declarations.

  1 | // File: C:\office\prepaid-insights-ui\app\layout.tsx
  2 | import * as entry from '../../../app/layout.js'
> 3 | import type { ResolvingMetadata } from 'next/dist/lib/metadata/types/metadata-interface.js'
    |                                        ^
  4 | 
  5 | type TEntry = typeof import('../../../app/layout.js')

I am using package ,"next": "^12.0.0",

**Comment by icyJoseph** [comment, marked as answer]
App Router is not supported in `next@12`. Are you trying to create an application using App Router? Or maybe by accident your project has a folder called app at the root? 

If you want to use App Router, you need next@13, for example `yarn add next@latest` will do the trick.

**Comment by okeegan** [comment]
I'm seeing this issue as well, are you able to provide more insight?

error:
```
../docs/types/app/layout.ts:3:59
Type error: Cannot find module 'next/dist/lib/metadata/types/metadata-interface.js' or its corresponding type declarations.

  1 | // File: /Users/okeegan/dev/xxx/yyy/web/src/app/layout.tsx
  2 | import * as entry from '../../../web/src/app/layout.js'
> 3 | import type { ResolvingMetadata, ResolvingViewport } from 'next/dist/lib/metadata/types/metadata-interface.js'
    |                                                           ^
  4 |
  5 | type TEntry = typeof import('../../../web/src/app/layout.js')

```

file: /src/app/layout.tsx
```
import type {Metadata} from 'next'
import {Montserrat} from 'next/font/google'
import './globals.css'

const montserrat = Montserrat({subsets: ['latin']})

export const metadata: Metadata = {
    title: 'zzz',
    description: 'xxx',
}

export default function RootLayout({ children }: { children: React.ReactNode }) {
    return (
        <html lang="en">
        <body className={montserrat.className}>{children}</body>
        </html>
    )
}
```

compiled layout.ts (export mode)
```
// File: /Users/okeegan/dev/xxx/yyy/web/src/app/layout.tsx
import * as entry from '../../../../../../../web/src/app/layout.js'
import type { ResolvingMetadata, ResolvingViewport } from 'next/dist/lib/metadata/types/metadata-interface.js'

type TEntry = typeof import('../../../../../../../web/src/app/layout.js')

( ... and so on ...)
```

Red squiggly under 'next/dist/lib/metadata/types/metadata-interface.js'
```
TS2307: Cannot find module
next/dist/lib/metadata/types/metadata-interface.js
or its corresponding type declarations.
```

Which checks out because looking at the directory structure of the built project:
**Note:** the built version is output _outside_ the `web` directory.
![image](https://github.com/vercel/next.js/assets/4075884/0f8d6f47-8c5d-4bf1-8609-c0dca481e195)

* There is no `next` directory at all.
* Changing the project to `standalone` had no effect.

Looks to me like the import used makes assumptions about the directory structure of the output. This is what my current structure looks like:
![image](https://github.com/vercel/next.js/assets/4075884/c40f86f3-3dd9-4166-b4fc-5bf819828655)

Versions
```
"next": "^14.0.4",
    "react": "^18",
    "react-dom": "^18"
```

Your help would be much appreciated. :)

**Reply by seheon99** [reply]
https://github.com/vercel/next.js/issues/60152
